modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -6411,7 +6411,14 @@ else if (newSysTtl == CU.TTL_ZERO) {
                 CacheLazyEntry<Object, Object> interceptEntry =
                     new CacheLazyEntry<>(cctx, entry.key, null, oldVal, null, keepBinary);
 
-                Object interceptorVal = cctx.config().getInterceptor().onBeforePut(interceptEntry, updated0);
+                Object interceptorVal = null;
+
+                try {
+                    interceptorVal = cctx.config().getInterceptor().onBeforePut(interceptEntry, updated0);
```

Comment:
How this change connected with the ticket?

Reply:
you are right, without this fix we could obtain erroneous CorruptedTreeException i want to fix it here too.

Reply:
Ok. Does here be exception logic?

Reply:
does question still actual ?